### PR TITLE
Support for New Quests

### PR DESCRIPTION
**Pull request by 123FLO321**
## Description
Support for New Quests

## How Has This Been Tested?
Untested

## Images
new images to items folder: https://cdn.discordapp.com/attachments/499498403487809537/750411111085375583/icons.zip

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by PoGoJoshS** [comment]
Mega energy quests are being displayed on the map with 0_i0.png instead of 0_i-6.png. 0_i-6.png has been generated in quest folder.

**Comment by 123FLO321** [comment]
@PoGoJoshS only Quest scanned after this pr sill show the correct icon.

**Comment by Bernie1410** [comment]
Works great

**Comment by imArize** [comment]
Mega energy quests are not showing on top of other quests when set to L or XL. This works for all other quests though.
Screenshot_1

Also, maybe show all energy quests when no pokemon are selected? I've had several user (including me) confused when all quests but the energy ones are disabled.

**Comment by 123FLO321** [comment]
Yes the pokemon Filter applying is a bug and will be fixed. However it requires a db migration.

**Comment by Fabio1988** [comment]
So energy is only visualized if the related pokemon and mega energy is enabled in filter? Only enable mega energy is not enough?

**Comment by drfsupercenter** [comment]
Functionally, this works fine and doesn't seem to break anything else.

However, there is a bug with FILTERING these quests - if you disable everything and enable Mega Energy, none of them show up unless you *also* enable Venusaur, Blastoise and/or Charizard.  That should not be required if you just want to see *any* mega energy quests.

File: Sources/RealDeviceMap/Modell/Pokestop.swift [hunk 1]
```diff
@@ -379,9 +379,16 @@ class Pokestop: JSONConvertibleObject, WebHookEvent, Hashable {
...
                 let info = rewardData.pokecoin
                 infoData["amount"] = info
+            case .sticker:
+                let info = rewardData.sticker
+                infoData["amount"] = info.amount
+                infoData["sticker_id"] = info.stickerID
+            case .megaResource:
+                let info = rewardData.megaResource
+                infoData["amount"] = info.amount
+                infoData["pokemon_id"] = info.pokemonID.rawValue
```
[the remark this answers is not in the document]
Comment (123FLO321) [reply Sources/RealDeviceMap/Modell/Pokestop.swift:394]: Fixed the incorrect trigger fixing this problem

**Review by kbtbc — DISMISSED** [review]
Running for a day w/out any issues.  -- oh, just realized I'm behind some commits.

**Comment by LordJavi** [comment]
Running lasts commits + migration. Working well so far

**Review by thermoscookies — APPROVED** [review]
working well for me

**Review by LordJavi — DISMISSED** [review]

**Review by ch1zzu — APPROVED** [review]

**Review by SkOODaT — APPROVED** [review]
seems good to merge, everything works as expected :), 
suggestion to add  `rewardString = 'i-6-' + info.pokemon_id` onto whatever things in index to allow for specific energy icons tho ..... 
ie: 
.........i2_i-6-6

sizeing would need to be corrected in index but thats a minor fix :)

**Review by SkOODaT — APPROVED** [review]

**Pull request merged**